src/components/v5/common/ActionSidebar/hooks/useGetStreamingPaymentData.ts

```diff
@@ -0,0 +1,63 @@
+import { useEffect, useState } from 'react';
+
+import { FAILED_LOADING_DURATION as POLLING_TIMEOUT } from '~frame/LoadingTemplate/index.ts';
+import { useGetStreamingPaymentQuery } from '~gql';
+import noop from '~utils/noop.ts';
+import { getSafePollingInterval } from '~utils/queries.ts';
+
+export const useGetStreamingPaymentData = (
+  streamingPaymentId: string | null | undefined,
+) => {
+  const pollInterval = getSafePollingInterval();
+
+  const { data, refetch, loading, startPolling, stopPolling } =
+    useGetStreamingPaymentQuery({
+      variables: {
+        streamingPaymentId: streamingPaymentId || '',
+      },
+      skip: Number.isNaN(streamingPaymentId),
+      fetchPolicy: 'network-only',
```

Comment:
Perhaps `cache-and-network` would work better here, to avoid making network calls more often than polling interval